### PR TITLE
feat: Add python example for appsync-graphql-dynamodb

### PR DESCRIPTION
**Pull request by zanhsieh**
Add python example for appsync-graphql-dynamodb
---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

**Comment by zanhsieh** [comment]
@rix0rrr
Could you review this? Travis seems stuck on something not related to my code.

File: python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    StreamViewType,
+    BillingMode
+)
+from aws_cdk.aws_iam import (
+    Role,
+    ServicePrincipal,
+    ManagedPolicy
+)
+
+class AppSyncCdkStack(core.Stack):
```
Comment (garnaat) [inline python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py]: PEP8 - A class definition should be preceded by two blank lines.

File: python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        table_name = 'items'
+
+        items_graphql_api = CfnGraphQLApi(
+            self, 'ItemsApi',
+            name='items-api',
+            authentication_type='API_KEY'
+        )
+
+        CfnApiKey(
+            self, 'ItemsApiKey',
```
Comment (garnaat) [inline python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py]: PEP8 - Trailing whitespace

File: python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            self, 'ItemsTable',
+            table_name=table_name,
+            partition_key=Attribute(
+                name=f'{table_name}Id',
+                type=AttributeType.STRING
+            ),
+            billing_mode=BillingMode.PAY_PER_REQUEST,
+            stream=StreamViewType.NEW_IMAGE,
+
+            ## The default removal policy is RETAIN, which means that cdk destroy will not attempt to delete
```
Comment (garnaat) [inline python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py]: PEP8 - Use a single hashmark, not two, for block comments.  Also, line length.

File: python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+            ## the new table, and it will remain in your account until manually deleted. By setting the policy to 
+            ## DESTROY, cdk destroy will delete the table (even if it has data in it) 
+            removal_policy=core.RemovalPolicy.DESTROY ## NOT recommended for production code
```
Comment (garnaat) [inline python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py]: PEP8 - Inline comments should be preceded by two spaces, not one.  Also, use a single hash mark.

File: python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+            self, 'ItemsDynamoDBRole',
+            assumed_by=ServicePrincipal('appsync.amazonaws.com')
+        )
+
+        items_table_role.add_managed_policy(ManagedPolicy.from_aws_managed_policy_name('arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess'))
+
+        data_source = CfnDataSource(
+            self, 'ItemsDataSource', 
+            api_id=items_graphql_api.attr_api_id,
+            name='ItemsDynamoDataSource',
```
Comment (garnaat) [inline python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py]: PEP8 - trailing white space

File: python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+                "{table_name}Id": $util.dynamodb.toDynamoDBJson($ctx.args.{table_name}Id)
+                }}
+            }}""",
+            response_mapping_template="$util.toJson($ctx.result)"
+        )
+
+        get_one_resolver.add_depends_on(api_schema)
+
+        get_all_resolver = CfnResolver(
+            self, 'GetAllQueryResolver',
```
Comment (garnaat) [inline python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py]: PEP8 - trailing white space

File: python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+                "limit": $util.defaultIfNull($ctx.args.limit, 20),
+                "nextToken": $util.toJson($util.defaultIfNullOrEmpty($ctx.args.nextToken, null))
+            }}""",
+            response_mapping_template="$util.toJson($ctx.result)"
+        )
+
+        get_all_resolver.add_depends_on(api_schema)
+
+        save_resolver = CfnResolver(
+            self, 'SaveMutationResolver',
```
Comment (garnaat) [inline python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py]: PEP8 - trailing white space

File: python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+                "operation": "DeleteItem",
+                "key": {{
+                "{table_name}Id": $util.dynamodb.toDynamoDBJson($ctx.args.{table_name}Id)
+                }}
+            }}""",
+            response_mapping_template="$util.toJson($ctx.result)"
+        )
+
+        delete_resolver.add_depends_on(api_schema)
+
```
Comment (garnaat) [inline python/appsync-graphql-dynamodb/app_sync_cdk/app_sync_cdk_stack.py]: PEP8 - trailing blank lines at end of file

**Comment by zanhsieh** [comment]
@garnaat 
Done. Please review again. Thank you.

**Comment by garnaat** [comment]
The Travis problem is unrelated to your PR.  I will try to fix that tomorrow.  Thanks!

**Comment by garnaat** [comment]
Hmm, my bad.  I merged without a README file.  Could you do a separate PR to add a README file that describes what this example does?

**Comment by zanhsieh** [comment]
@garnaat 
Sure
https://github.com/aws-samples/aws-cdk-examples/pull/124

**Pull request merged**